app/assets/javascripts/tolk/actions.js

```diff
@@ -13,30 +13,48 @@ $(function () {
   });
 
   // Google Translate action (NEW)
-  $('.gtranslate').click(function (e) {
+  $('.js-gtranslate-single').click(function (e) {
     e.preventDefault();
 
     var origText = $(this).parent(".actions").next('.original').find("textarea").val();
     var destLang = $(this).data('locale');
     if(destLang === 'es-CO') { destLang = 'es'; }
     var self = this;
-    gapi.client.init({
-      'apiKey': window.googleApiKey,
-      'discoveryDocs': ['https://www.googleapis.com/discovery/v1/apis/translate/v2/rest'],
-    }).then(function() {
-      return gapi.client.language.translations.list({
-        q: origText,
-        source: 'en',
-        target: destLang,
-      });
-    }).then(function(response) {
+    gTranslate(origText, destLang).then(function(response) {
       var destText = response.result.data.translations[0].translatedText;
       $(self).parents('tr').find(".translation textarea").val(destText);
     }, function(reason) {
       console.warn('Error: ' + reason.result.error.message);
     });
   });
 
+  $('.js-gtranslate-all').click(function (e) {
+    e.preventDefault();
+    var destLang = $(this).data('locale');
+    if(destLang === 'es-CO') { destLang = 'es'; }
+
+    var tbody = $("form.edit_locale").find("tbody")
+    var texts = [];
+    tbody.find("td.original").each(function() {
+      texts.push($(this).find("textarea").val());
+    })
+
+    gTranslate(texts, destLang).then(function(response) {
+      var translations = response.result.data.translations;
+      tbody.find("td.translation textarea").each(function(index) {
+        var value = translations[index].translatedText;
+        if($(this).val().length > 0 ||
+          value.indexOf("% {") > -1 ||
```

Comment:
I think it would be safer if this combination exists in origin text, not translated text.
Also shouldn't it be `%{` instead of `% {`?

Reply:
Actually, it was somehow adding space here. Maybe I can search for just `%`
API is not returning original text sadly, but we can fetch it from DOM, but it required some additional manipulation.